sunpy/data/_sample.py

```diff
@@ -83,20 +83,23 @@ def download_sample_data(progress=True, overwrite=True):
         for base_url in _base_urls:
             full_file_name = file_name[0] + file_name[1]
             if url_exists(os.path.join(base_url, full_file_name)):
-                f = download_file(os.path.join(base_url, full_file_name))
-                real_name, ext = os.path.splitext(full_file_name)
-
-                if file_name[1] == '.zip':
-                    print("Unpacking: {}".format(real_name))
-                    with ZipFile(f, 'r') as zip_file:
-                        zip_file.extract(real_name, sampledata_dir)
-                    os.remove(f)
-                else:
-                    # move files to the data directory
-                    move(f, os.path.join(sampledata_dir, file_name[0]))
+                try:
+                    f = download_file(os.path.join(base_url, full_file_name))
+                    real_name, ext = os.path.splitext(full_file_name)
+
+                    if file_name[1] == '.zip':
+                        print("Unpacking: {}".format(real_name))
+                        with ZipFile(f, 'r') as zip_file:
+                           zip_file.extract(real_name, sampledata_dir)
+                        os.remove(f)
+                    else:
+                        # move files to the data directory
+                        move(f, os.path.join(sampledata_dir, file_name[0]))
                 # increment the number of files obtained to check later
-                number_of_files_fetched += 1
-                break
+                    number_of_files_fetched += 1
+                    break
+                except socket.timeout as e:
```

Comment:
@abit2 you don't want to `raise` warnings, because that raises them as errors, you want to use [`warnins.warn`](https://docs.python.org/2/library/warnings.html#warnings.warn). Also you probably want to pass along the exception as well something like:

```python
warnings.warn("Download failed with error {}. \n Retrying with different mirror.".format(e))
```